mitiq/ddd/rules/rules.py

```diff
@@ -22,22 +22,31 @@
 from itertools import cycle
 
 
-def construct_rule(
-    slack_length: int, spacing: int, gates: List[Gate]
+def general_rule(
+    slack_length: int, gates: List[Gate], spacing: int = -1
 ) -> Circuit:
     """Returns a digital dynamical decoupling sequence, based on inputs.
 
     Args:
         slack_length: Length of idle window to fill.
         spacing: How many identity spacing gates to apply between dynamical
-            decoupling gates, as a non-negative int. Defaults to evenly
-            spaced.
+            decoupling gates, as a non-negative int. Negative int corresponds
+            to default. Defaults to maximal spacing that fits a single sequence
+            in the given slack window.
+            E.g. given slack_length = 20, gates = [X, X] the spacing defaults
+            to 6 and the rule returns the sequence:
+            ──I──I──I──I──I──I──X──I──I──I──I──I──I──X──I──I──I──I──I──I──
+            given slack_length = 20, gates [X, Y, X, Y] the spacing defaults
+            to 3 and the rule returns the sequence:
+            ──I──I──I──I──X──I──I──I──Y──I──I──I──X──I──I──I──Y──I──I──I──
         gates: A list of Cirq gates to build the rule. E.g. [X, X] is the xx
             sequence, [X, Y, X, Y] is the xyxy sequence
             - Note: To repeat the sequence, specify a repeated gateset
     Returns:
         A digital dynamical decoupling sequence, as a cirq circuit
     """
+    if slack_length < 2 or slack_length < len(gates):
```

Comment:
I realized some potential errors that we should probably check:
- Error if gates are not single-qubit gates. I think this is important.
- (Maybe) Error if `cirq.unitary(Circuit(gates))` is not np.close to the identity matrix. 
- (Maybe) Error if `len(gates) < 2` , unless we want to support the limit case `gates=[ I ]`.

Reply:
Currently the error thrown when a multi-qubit gate is provided as input bubbles up from `cirq` (because we assume a single qubit):
`[ValueError: Wrong number of qubits for <cirq.CNOT>. Expected 2 qubits but got <[cirq.LineQubit(0)]>.]()`

Do you think that's sufficient? I think that's fine, though we might want to check that the gate(s) don't have qubit assignment. Currently the error thrown is `[AttributeError: 'GateOperation' object has no attribute 'on']()` when a `GateOperation` is passed to a rule (in the 1 or 2 qubit gate case).